websocket_server.go

```diff
@@ -346,15 +346,29 @@ func (n *pushServer) start(ctx context.Context) {
 					}
 
 					if n.cfg.pushServer.dispatchHandler != nil {
-						ok, err := n.cfg.pushServer.dispatchHandler.ShouldDispatch(session, event, eventSummary)
+						dispatch, err := n.cfg.pushServer.dispatchHandler.ShouldDispatch(session, event, eventSummary)
 						if err != nil {
+							//if 'ShouldDispatch' returns an error type that has the method 'ShouldCloseSocket() (bool, int)'
+							//then it can inform whether the socket should be closed along with a code to use in the
+							//close message. This is useful in cases where it makes no sense to continue serving the client
+							//(e.g. the client configured a semantically invalid push config, for example: they used a comparator
+							//that is not yet supported/implemented).
+							if closer, ok := err.(interface {
```

Comment:
can we define this in the interface? I know it's a breaking change, but that's ok. this makes the api a bit unclear.

Reply:
I misread that a bit. it's not a breaking change, however it would be cool if this interface was defined

Reply:
donezo